### PR TITLE
Feature(multiple pop): now we can pop multipe scenes with Actions.pop

### PR DESCRIPTION
**Pull request by Swordsman-Inaction**
now we can pop multipe scenes with Actions.pop([number]), like Actions.pop(2)

But I don't know much about the mechanism, please tell me if I miss something.

@aksonov

### CODE REVIEW

File: src/Reducer.js [hunk 1]
```diff
@@ -63,11 +63,21 @@ function inject(state, action, props, scenes) {
...
         return state;
       }
+
+      let popNum = 1
+      if (action.data) {
+        assert(typeof(action.data) === 'number', 'The data is the number of scenes you want to pop, it must be Number')
+        popNum = action.data
+        assert(popNum % 1 === 0, 'The data is the number of scenes you want to pop, it must be integer.')
+        assert(popNum > 1, 'The data is the number of scenes you want to pop, it must be bigger than 1.')
+        assert(popNum < state.children.length, "The data is the number of scenes you want to pop, it must be smaller than scenes stack's length.")
```
Comment (aksonov) [inline src/Reducer.js]: Probably you need to check against state.index, not state.children.length...

Comment (Swordsman-Inaction) [inline src/Reducer.js]: @aksonov  changed

**Comment by Swordsman-Inaction** [comment]
@aksonov In the `Reducer.js`, before `update` is called, `action.parent` will be assigned, But neither `update()` or `inject` use it. So does it matter? If it matters, I think I should update the implement, since the `action.parent` is wrong if you are popping several scenes.

**Comment by aksonov** [comment]
@Swordsman-Inaction it doesn't matter actually, if your PR changes state properly without settings action.parent. Could you fix conflicts and eslint errors?

**Comment by joenoon** [comment]
I think this can be closed since https://github.com/aksonov/react-native-router-flux/pull/798 was merged?

**Comment by Swordsman-Inaction** [comment]
@joenoon @aksonov  #798 is a good feature. But this one is trying to get this (https://github.com/aksonov/react-native-router-flux/issues/39#issuecomment-157990785) back.
Both of them are basically trying to do the same thing. If you guys think #798 is enough, then this one can be closed

**Comment by joenoon** [comment]
@Swordsman-Inaction ok then I can see how this makes sense.  Besides the eslint and conflicts, the only other thing that comes to mind is using `action.data`.  Should that be so generic, or should it be `action.popNum` or something?

**Comment by Swordsman-Inaction** [comment]
@joenoon `action.popNum` looks great. I'll fix those

**Comment by Swordsman-Inaction** [comment]
@aksonov @joenoon Done

**Pull request merged**